test/Rules/highestRollingPlayerGoesFirst.test.js

```diff
@@ -0,0 +1,80 @@
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const highestRollingPlayerGoesFirst = require('../../entities/Rules/highestRollingPlayerGoesFirst');
+const Dice = require('../../entities/Components/Dice');
+const { createPlayer } = require('../testutils');
+const mockUIFactory = require('../mocks/UI');
+
+describe('Rules -> highestRollingPlayerGoesFirst', () => {
+  let userInterface;
+  let input;
+  let diceStub;
+
+  beforeEach(() => {
+    userInterface = mockUIFactory();
+    input = {
+      players: [
+        createPlayer({ name: 'Player1' }),
+        createPlayer({ name: 'Player2' }),
+      ],
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it(`should assign indices to each player`, () => {
+    highestRollingPlayerGoesFirst({ UI: userInterface }, input);
+    for (let index = 0; index < input.players.length; index++) {
```

Comment:
```javascript
input.players.forEach((player, index) => expect(
    player.id === index,
    `Player index of ${ index } does not match player position`
));
```

Reply:
I think it might be better to do a `.map`, though, and then an `every` ?
```javascript
expect(
    input.players.map(player => player.id).every((playerId, index) => playerId === index) === true,
   `All players are not matched with their index`
);
```

Reply:
Possibly taking that middle logic out, though, and putting it a variable

Reply:
No longer applicable.